fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/impl/registry/sync/RegistrySyncManager.java

```diff
@@ -168,18 +204,38 @@ public static CompoundTag toTag(boolean isClientSync) {
 					registryTag.putInt(id.toString(), rawId);
 				}
 
+				// Look for existing registry key/values that are not in the current registries
+				if (!isClientSync && existingRegistryData != null) {
+					for (String key : existingRegistryData.getKeys()) {
+						if (!registryTag.contains(key)) {
+							LOGGER.info("Saving orphaned registry entry: " + key);
+							registryTag.putInt(key, registryTag.getInt(key));
+						}
+					}
+				}
+
 				mainTag.put(registryId.toString(), registryTag);
 			}
 		}
 
+		// Ensure any orphaned registry's are kept on disk
+		if (!isClientSync && activeIdMap != null) {
+			for (String registryKey : activeIdMap.getKeys()) {
+				if (!mainTag.contains(registryKey)) {
+					LOGGER.info("Saving orphaned registry: " + registryKey);
+					mainTag.put(registryKey, activeIdMap.getCompound(registryKey));
+				}
+			}
+		}
+
 		CompoundTag tag = new CompoundTag();
 		tag.putInt("version", 1);
```

Comment:
Would a version bump make sense?

Obviously the schema hasn't changed but it's worth considering.

Reply:
imo using an arbitrary verison here is not as good as using channels with version in channel names, like `fabric-registry-sync:sync/v1` etc

Reply:
I think once im more happy with the state of the PR I will make it a -v1, its easier to see the changes like this now.

Reply:
Decided this is not needed, as it should be forwards/backwards compatible, as the format didnt change.